hazelcast/src/main/java/com/hazelcast/config/PermissionConfig.java

```diff
@@ -199,6 +231,15 @@ public static PermissionType getType(String nodeName) {
             return null;
         }
 
+        public static PermissionType getTypeByPermissionClassName(String permissionClassname) {
+            for (PermissionType type : PermissionType.values()) {
```

Comment:
Would it not be better to make a `Map<String, PermissionType>` instead?

Reply:
This way I don't need to care about thread safety. And we already have the same approach in the `getType()` method.

Reply:
> This way I don't need to care about thread safety.

If it was a static lookup it'd be ok anyway?

> And we already have the same approach in the `getType()` method.

Good point. Maybe both should change together.

Reply:
I've simplified the code, but I didn't introduce additional data structure for the reverse lookups. IMO the enum size is small enough to be OK to do scan-through on the lookups.